packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -165,9 +181,24 @@ class CodeSizeController {
 
       diffMap = compareProgramInfo(oldApkProgramInfo, newApkProgramInfo);
     } else {
-      diffMap = buildComparisonTreemap(oldFile.data, newFile.data);
+      try {
+        diffMap = buildComparisonTreemap(oldFile.data, newFile.data);
+      } catch (error) {
+        onError('Failed to load snapshot: File type not supported.');
```

Comment:
"Failed to load diff: file type not supported. 

The code size tool supports AOT v8 snapshots, instruction sizes, and apk-analysis files. See documentation for how to generate these files."

Reply:
Done